app/frontend/components/pages/txHistory/transactionHistory.tsx

```diff
@@ -153,68 +158,123 @@ const ExportCSV = ({transactionHistory, stakingHistory}: Props): h.JSX.Element =
     type: TxSummaryType
     txHash?: string
     dateTime: moment.Moment
-    sent?: Lovelace
-    received?: Lovelace
     fee?: Lovelace
-  }
+    currency: string
+  } & (
+    | {
+        assetFamily: AssetFamily.ADA
+        sent?: Lovelace
+        received?: Lovelace
+      }
+    | {
+        assetFamily: AssetFamily.TOKEN
+        sent?: number
+        received?: number
+        decimals: number
+      }
+  )
 
-  const transactionsEntries: Array<TxEntry> = transactionHistory.map(
-    (transaction: TxSummaryEntry) => {
+  const transactionsEntries: Array<TxEntry> = transactionHistory.flatMap(
+    (transaction: TxSummaryEntry): Array<TxEntry> => {
       const common = {
         txHash: transaction.ctbId,
         dateTime: moment.utc(new Date(transaction.ctbTimeIssued * 1000)),
       }
 
+      const createTokenEntries = (): Array<TxEntry> =>
+        transaction.tokenEffects.map((tokenEffect: Token) => {
+          const tokenMetadata = tokensMetadata.get(
+            createTokenRegistrySubject(tokenEffect.policyId, tokenEffect.assetName)
+          )
+          return {
+            ...common,
+            ...(tokenEffect.quantity > 0
+              ? {
+                type: TxSummaryType.RECEIVED,
+                received: tokenEffect.quantity,
+              }
+              : {
+                type: TxSummaryType.SENT,
+                sent: Math.abs(tokenEffect.quantity),
+              }),
+            assetFamily: AssetFamily.TOKEN,
+            currency:
+              tokenMetadata?.ticker ||
+              encodeAssetFingerprint(tokenEffect.policyId, tokenEffect.assetName),
```

Comment:
Since tickers are not unique, I might be nice to have it as `ticker ? ticker (fingerprint) : fingerprint` . Just a nitpick, most likely this wont cause any issues but its a bit safer this way IMHO, what do you think?